src/phase0/state_transition/epoch_processing.rs

```diff
@@ -154,8 +156,63 @@ pub fn process_registry_updates<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    // Process activation eligibility and ejections
+    let current_epoch = get_current_epoch(state, context);
+
+    for i in 0..state.validators.len() {
+        let validator = &mut state.validators[i];
+        if is_eligible_for_activation_queue(validator, context) {
+            validator.activation_eligibility_epoch = current_epoch + 1;
+        }
+
+        if is_active_validator(validator, current_epoch)
+            && validator.effective_balance <= context.ejection_balance
+        {
+            initiate_validator_exit(state, i, context);
+        }
+    }
+
+    // Queue validators eligible for activation and not yet dequeued for activation
+    let _activation_queue = state
+        .validators
+        .iter()
+        .enumerate()
+        .filter(|&(_, validator)| is_eligible_for_activation(state, validator))
+        .collect::<Vec<(ValidatorIndex, &Validator)>>()
+        // Order by the sequence of activation_eligibility_epoch setting and then index
+        .sort_by(|a, b| {
+            let a_activation_eligibility_epoch = a.1.activation_eligibility_epoch;
+            let a_index = a.0;
+            let b_activation_eligibility_epoch = b.1.activation_eligibility_epoch;
+            let b_index = b.0;
+            match a_activation_eligibility_epoch.cmp(&b_activation_eligibility_epoch) {
+                cmp::Ordering::Less => cmp::Ordering::Less,
+                cmp::Ordering::Greater => cmp::Ordering::Greater,
+                cmp::Ordering::Equal => match a_index.cmp(&b_index) {
+                    cmp::Ordering::Less => cmp::Ordering::Less,
+                    cmp::Ordering::Greater => cmp::Ordering::Greater,
+                    // TODO: throw some error if the valators' indices are equal -- should not be possible
+                    cmp::Ordering::Equal => cmp::Ordering::Equal,
+                },
+            }
```

Comment:
so i think we can simplify all of this by taking advantage of the existing impl of `PartialOrd` for tuples...

something like:

```rust
.sort_by(|(a_index, a), (b_index, b)| { (a.epoch, a_index) < (b.epoch, b_index) })
```

where you can use tuple destructuring in the closure argument to save some typing